### PR TITLE
Indexing may panic because of 'nil' slice

### PR DESCRIPTION
**Pull request by Akiros001**

### CODE REVIEW

File: kinder/pkg/cluster/manager/manage.go [hunk 1]
```diff
@@ -151,9 +151,11 @@ func (c *ClusterManager) CopyFile(source, target string) error {
...
 
 	if targetNodes == nil {
-		fmt.Printf("Copying from %s ...\n", sourceNodes[0].Name())
-		if err := sourceNodes[0].CopyFrom(sourcePath, targetPath); err != nil {
-			return err
+		if sourceNodes != nil {
+			fmt.Printf("Copying from %s ...\n", sourceNodes[0].Name())
+			if err := sourceNodes[0].CopyFrom(sourcePath, targetPath); err != nil {
+				return err
+			}
```
Comment (neolit123) [inline kinder/pkg/cluster/manager/manage.go:158]: we have a `sourceNodes != nil` check above, so i don't think this is exactly needed.

note: you need sign the CLA to contribute to k8s projects.

**Comment by pacoxu** [comment]
Can you reproduce the problem and provide the log?

File: kinder/pkg/cluster/manager/manage.go (hunk 1 above)
Comment (neolit123) [reply kinder/pkg/cluster/manager/manage.go:158]: the logic here is the following in pseudo code:

```
if sourceNodes == nil && targetNodes == nil { return error }
if sourceNodes != nil && targetNodes != nil { return error }
```

so at this point either sourceNodes or targetNodes is nil, but no both.

/close

**Pull request closed without merging**